Controllers/conversationController.js

```diff
@@ -0,0 +1,141 @@
+const CatchAsyncError = require('../Utils/CatchAsyncError')
+const db = require('../db/models/index')
+const User = db.User
+const UserConversation = db.UserConversation
+const Messages = db.Message
+const { Op } = require('sequelize')
+
+// WORK IN PROGRESS
+exports.createConversation = CatchAsyncError(async (req, res, next) => {
+  const searchedUserUsername = req.body.searchedUserUsername
+  const currentUserUsername = req.body.currentUserUsername
+
+  const searchedUserArr = await User.findAll({
+    where: {
+      username: searchedUserUsername
+    }
+  })
+  const currentUserArr = await User.findAll({
+    where: {
+      username: currentUserUsername
+    }
+  })
+  const currentUser = currentUserArr[0].dataValues
+  const searchedUser = searchedUserArr[0].dataValues
+
+  const currentUserConversations = await UserConversation.findAll({
+    attributes: ['ConversationId'],
+    where: {
+      UserId: currentUser.id
+    }
+  })
+  const containerCurrentUser = []
+
+  currentUserConversations.forEach(element => {
+    const [num] = (Object.values(element.dataValues))
+    containerCurrentUser.push(num)
+  })
```

Comment:
Instead of mapping the conversation ids like this, you can use the `raw` option in the sequelize query.
like this:
`const currentUserConversations = await UserConversation.findAll({
    attributes: ['ConversationId'],
    where: {
      UserId: currentUser.id
    },
    raw: true
  })`
This will return a simple array instead of an array of Sequelize objects.